src/Platforms/AbstractPlatform.php

```diff
@@ -1575,40 +1576,21 @@ public function getColumnDeclarationSQL(string $name, array $column): string
      * Returns the SQL snippet that declares a floating point column of arbitrary precision.
      *
      * @param mixed[] $column
+     *
+     * @throws ColumnPrecisionRequired
+     * @throws ColumnScaleRequired
      */
     public function getDecimalTypeDeclarationSQL(array $column): string
     {
-        if (empty($column['precision'])) {
-            if (! isset($column['precision'])) {
-                Deprecation::trigger(
-                    'doctrine/dbal',
-                    'https://github.com/doctrine/dbal/pull/5637',
-                    'Relying on the default decimal column precision is deprecated'
-                        . ', specify the precision explicitly.',
-                );
-            }
-
-            $precision = 10;
-        } else {
-            $precision = $column['precision'];
+        if (! isset($column['precision'])) {
+            throw ColumnPrecisionRequired::new();
```

Comment:
Is this typically going to be caught and improved with additional context elsewhere, or should we add additional context based on information contained in `$column`?

Reply:
I got mislead by the variable name (which should probably be `$columnOptions`, I don't think there is actually going to be anything helpful in there. The concern still stands though: are these exceptions going to be handled at a level where the column's fully qualified name is available?

Reply:
I understand the problem. We have the same problem in other places, e.g. https://github.com/doctrine/dbal/blob/08630c6188d70b0f7a98b9c4a079f5f7a76b6cba/src/Platforms/AbstractPlatform.php#L267-L271

And it's even more acute there since the column name is not available in the scope from where the exception is thrown.

I cannot think of a simple solution at this point. I don't want to introduce a dependency on `$column['name']` just to use it as a parameter for the exception because the exception is unrelated to the name.

A proper solution might be to improve the exception hierarchy:

1. All these three exceptions should have a common type (e.g. `InvalidColumnType`).
2. The code that calls the functions that may throw such exceptions should have the column name in its scope and should catch these exceptions and throw a more user-friendly exception (e.g. `InvalidColumnDefinition`).